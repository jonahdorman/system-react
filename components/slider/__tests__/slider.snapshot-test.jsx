/* eslint-env jest */
import React from 'react';
import { renderMarkup, testDOMandHTML } from '../../../tests/snapshot-helpers';

import BaseExample from '../__examples__/base';
import DisabledExample from '../__examples__/disabled';
import ErrorExample from '../__examples__/error';
import SizesExample from '../__examples__/sizes';
import VerticalExample from '../__examples__/vertical';

testDOMandHTML({
	name: 'Base label, no label, min/max, min/max/step',
	test,
	Component: BaseExample,
});

testDOMandHTML({
	name: 'Disabled',
	test,
	Component: DisabledExample,
});

testDOMandHTML({
	name: 'Error',
	test,
	Component: ErrorExample,
});

testDOMandHTML({
	name: 'Sizes',
	test,
	Component: SizesExample,
});

testDOMandHTML({
	name: 'Vertical',
	test,
	Component: VerticalExample,
});
