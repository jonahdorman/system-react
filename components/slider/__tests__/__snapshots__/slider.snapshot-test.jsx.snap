// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base label, no label, min/max, min/max/step DOM Snapshot 1`] = `
<section>
  <ol>
    <li
      className="slds-p-bottom_large"
    >
      <h1
        className="slds-text-title_caps slds-p-vertical_medium"
      >
        1. Base Input with label
      </h1>
      <div
        className="slds-form-element"
      >
        <label
          className="slds-form-element__label"
          htmlFor="base-id"
        >
          <span
            className="slds-slider-label"
          >
            <span
              className="slds-slider-label__label"
            >
              My Label
            </span>
            <span
              className="slds-slider-label__range"
            >
              0
               - 
              100
            </span>
          </span>
        </label>
        <div
          className="slds-form-element__control"
        >
          <div
            className="slds-slider"
          >
            <input
              className="slds-slider__range"
              id="base-id"
              max={100}
              min={0}
              onChange={[Function]}
              onInput={[Function]}
              step={1}
              type="range"
              value={0}
            />
            <span
              aria-hidden="true"
              className="slds-slider__value"
            >
              0
            </span>
          </div>
        </div>
      </div>
    </li>
    <li
      className="slds-p-bottom_large"
    >
      <h1
        className="slds-text-title_caps slds-p-vertical_medium"
      >
        2. Base Input with no label (assistive text)
      </h1>
      <div
        className="slds-form-element"
      >
        <label
          className="slds-form-element__label slds-assistive-text"
          htmlFor="assistiveText-id"
        >
          <span
            className="slds-slider-label"
          >
            <span
              className="slds-slider-label__label"
            >
              My Label
            </span>
            <span
              className="slds-slider-label__range"
            >
              0
               - 
              100
            </span>
          </span>
        </label>
        <div
          className="slds-form-element__control"
        >
          <div
            className="slds-slider"
          >
            <input
              className="slds-slider__range"
              id="assistiveText-id"
              max={100}
              min={0}
              onChange={[Function]}
              onInput={[Function]}
              step={1}
              type="range"
              value={0}
            />
            <span
              aria-hidden="true"
              className="slds-slider__value"
            >
              0
            </span>
          </div>
        </div>
      </div>
    </li>
    <li
      className="slds-p-bottom_large"
    >
      <h1
        className="slds-text-title_caps slds-p-vertical_medium"
      >
        3. Base Input with min and max.
      </h1>
      <div
        className="slds-form-element"
      >
        <label
          className="slds-form-element__label"
          htmlFor="min-max-id"
        >
          <span
            className="slds-slider-label"
          >
            <span
              className="slds-slider-label__label"
            >
              My Label
            </span>
            <span
              className="slds-slider-label__range"
            >
              0
               - 
              400
            </span>
          </span>
        </label>
        <div
          className="slds-form-element__control"
        >
          <div
            className="slds-slider"
          >
            <input
              className="slds-slider__range"
              id="min-max-id"
              max={400}
              min={0}
              onChange={[Function]}
              onInput={[Function]}
              step={1}
              type="range"
              value={200}
            />
            <span
              aria-hidden="true"
              className="slds-slider__value"
            >
              200
            </span>
          </div>
        </div>
      </div>
    </li>
    <li
      className="slds-p-bottom_large"
    >
      <h1
        className="slds-text-title_caps slds-p-vertical_medium"
      >
        4. Base Input with min, max and step.
      </h1>
      <div
        className="slds-form-element"
      >
        <label
          className="slds-form-element__label"
          htmlFor="min-max-step-id"
        >
          <span
            className="slds-slider-label"
          >
            <span
              className="slds-slider-label__label"
            >
              My Label
            </span>
            <span
              className="slds-slider-label__range"
            >
              0
               - 
              400
            </span>
          </span>
        </label>
        <div
          className="slds-form-element__control"
        >
          <div
            className="slds-slider"
          >
            <input
              className="slds-slider__range"
              id="min-max-step-id"
              max={400}
              min={0}
              onChange={[Function]}
              onInput={[Function]}
              step={100}
              type="range"
              value={200}
            />
            <span
              aria-hidden="true"
              className="slds-slider__value"
            >
              200
            </span>
          </div>
        </div>
      </div>
    </li>
  </ol>
</section>
`;

exports[`Base label, no label, min/max, min/max/step HTML Snapshot 1`] = `
"<section>
  <ol>
    <li class=\\"slds-p-bottom_large\\">
      <h1 class=\\"slds-text-title_caps slds-p-vertical_medium\\">1. Base Input with label</h1>
      <div class=\\"slds-form-element\\"><label class=\\"slds-form-element__label\\" for=\\"base-id\\"><span class=\\"slds-slider-label\\"><span class=\\"slds-slider-label__label\\">My Label</span><span class=\\"slds-slider-label__range\\">0 - 100</span></span></label>
        <div class=\\"slds-form-element__control\\">
          <div class=\\"slds-slider\\"><input type=\\"range\\" id=\\"base-id\\" class=\\"slds-slider__range\\" min=\\"0\\" max=\\"100\\" step=\\"1\\" value=\\"0\\" /><span class=\\"slds-slider__value\\" aria-hidden=\\"true\\">0</span></div>
        </div>
      </div>
    </li>
    <li class=\\"slds-p-bottom_large\\">
      <h1 class=\\"slds-text-title_caps slds-p-vertical_medium\\">2. Base Input with no label (assistive text)</h1>
      <div class=\\"slds-form-element\\"><label class=\\"slds-form-element__label slds-assistive-text\\" for=\\"assistiveText-id\\"><span class=\\"slds-slider-label\\"><span class=\\"slds-slider-label__label\\">My Label</span><span class=\\"slds-slider-label__range\\">0 - 100</span></span></label>
        <div class=\\"slds-form-element__control\\">
          <div class=\\"slds-slider\\"><input type=\\"range\\" id=\\"assistiveText-id\\" class=\\"slds-slider__range\\" min=\\"0\\" max=\\"100\\" step=\\"1\\" value=\\"0\\" /><span class=\\"slds-slider__value\\" aria-hidden=\\"true\\">0</span></div>
        </div>
      </div>
    </li>
    <li class=\\"slds-p-bottom_large\\">
      <h1 class=\\"slds-text-title_caps slds-p-vertical_medium\\">3. Base Input with min and max.</h1>
      <div class=\\"slds-form-element\\"><label class=\\"slds-form-element__label\\" for=\\"min-max-id\\"><span class=\\"slds-slider-label\\"><span class=\\"slds-slider-label__label\\">My Label</span><span class=\\"slds-slider-label__range\\">0 - 400</span></span></label>
        <div class=\\"slds-form-element__control\\">
          <div class=\\"slds-slider\\"><input type=\\"range\\" id=\\"min-max-id\\" class=\\"slds-slider__range\\" min=\\"0\\" max=\\"400\\" step=\\"1\\" value=\\"200\\" /><span class=\\"slds-slider__value\\" aria-hidden=\\"true\\">200</span></div>
        </div>
      </div>
    </li>
    <li class=\\"slds-p-bottom_large\\">
      <h1 class=\\"slds-text-title_caps slds-p-vertical_medium\\">4. Base Input with min, max and step.</h1>
      <div class=\\"slds-form-element\\"><label class=\\"slds-form-element__label\\" for=\\"min-max-step-id\\"><span class=\\"slds-slider-label\\"><span class=\\"slds-slider-label__label\\">My Label</span><span class=\\"slds-slider-label__range\\">0 - 400</span></span></label>
        <div class=\\"slds-form-element__control\\">
          <div class=\\"slds-slider\\"><input type=\\"range\\" id=\\"min-max-step-id\\" class=\\"slds-slider__range\\" min=\\"0\\" max=\\"400\\" step=\\"100\\" value=\\"200\\" /><span class=\\"slds-slider__value\\" aria-hidden=\\"true\\">200</span></div>
        </div>
      </div>
    </li>
  </ol>
</section>"
`;

exports[`Disabled DOM Snapshot 1`] = `
<div
  className="slds-form-element"
>
  <label
    className="slds-form-element__label"
    htmlFor="disabled-id"
  >
    <span
      className="slds-slider-label"
    >
      <span
        className="slds-slider-label__label"
      >
        Slider Label
      </span>
      <span
        className="slds-slider-label__range"
      >
        0
         - 
        100
      </span>
    </span>
  </label>
  <div
    className="slds-form-element__control"
  >
    <div
      className="slds-slider"
    >
      <input
        className="slds-slider__range"
        disabled={true}
        id="disabled-id"
        max={100}
        min={0}
        onChange={[Function]}
        onInput={[Function]}
        step={1}
        type="range"
      />
      <span
        aria-hidden="true"
        className="slds-slider__value"
      >
        0
      </span>
    </div>
  </div>
</div>
`;

exports[`Disabled HTML Snapshot 1`] = `
"<div class=\\"slds-form-element\\"><label class=\\"slds-form-element__label\\" for=\\"disabled-id\\"><span class=\\"slds-slider-label\\"><span class=\\"slds-slider-label__label\\">Slider Label</span><span class=\\"slds-slider-label__range\\">0 - 100</span></span></label>
  <div class=\\"slds-form-element__control\\">
    <div class=\\"slds-slider\\"><input type=\\"range\\" id=\\"disabled-id\\" class=\\"slds-slider__range\\" min=\\"0\\" max=\\"100\\" step=\\"1\\" disabled=\\"\\" /><span class=\\"slds-slider__value\\" aria-hidden=\\"true\\">0</span></div>
  </div>
</div>"
`;

exports[`Error DOM Snapshot 1`] = `
<div
  className="slds-form-element slds-has-error"
>
  <label
    className="slds-form-element__label"
    htmlFor="error-id"
  >
    <span
      className="slds-slider-label"
    >
      <span
        className="slds-slider-label__label"
      >
        Slider Label
      </span>
      <span
        className="slds-slider-label__range"
      >
        0
         - 
        100
      </span>
    </span>
  </label>
  <div
    className="slds-form-element__control"
  >
    <div
      className="slds-slider"
    >
      <input
        aria-describedby="error message"
        className="slds-slider__range"
        id="error-id"
        max={100}
        min={0}
        onChange={[Function]}
        onInput={[Function]}
        step={1}
        type="range"
      />
      <span
        aria-hidden="true"
        className="slds-slider__value"
      >
        0
      </span>
    </div>
    <div
      className="slds-form-element__help"
      id="error message"
    >
      There is a problem with this field
    </div>
  </div>
</div>
`;

exports[`Error HTML Snapshot 1`] = `
"<div class=\\"slds-form-element slds-has-error\\"><label class=\\"slds-form-element__label\\" for=\\"error-id\\"><span class=\\"slds-slider-label\\"><span class=\\"slds-slider-label__label\\">Slider Label</span><span class=\\"slds-slider-label__range\\">0 - 100</span></span></label>
  <div class=\\"slds-form-element__control\\">
    <div class=\\"slds-slider\\"><input type=\\"range\\" id=\\"error-id\\" class=\\"slds-slider__range\\" min=\\"0\\" max=\\"100\\" step=\\"1\\" aria-describedby=\\"error message\\" /><span class=\\"slds-slider__value\\" aria-hidden=\\"true\\">0</span></div>
    <div id=\\"error message\\" class=\\"slds-form-element__help\\">There is a problem with this field</div>
  </div>
</div>"
`;

exports[`Sizes DOM Snapshot 1`] = `
<div
  className="slds-grid slds-grid_pull-padded slds-grid_vertical-align-center"
>
  <div
    className="slds-col_padded"
  >
    <div
      className="slds-form-element"
    >
      <label
        className="slds-form-element__label"
        htmlFor="x-small-id"
      >
        <span
          className="slds-slider-label"
        >
          <span
            className="slds-slider-label__label"
          >
            Slider Label
          </span>
          <span
            className="slds-slider-label__range"
          >
            0
             - 
            100
          </span>
        </span>
      </label>
      <div
        className="slds-form-element__control"
      >
        <div
          className="slds-slider slds-size_x-small"
        >
          <input
            className="slds-slider__range"
            id="x-small-id"
            max={100}
            min={0}
            onChange={[Function]}
            onInput={[Function]}
            step={1}
            type="range"
          />
          <span
            aria-hidden="true"
            className="slds-slider__value"
          >
            0
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="slds-col_padded"
  >
    <div
      className="slds-form-element"
    >
      <label
        className="slds-form-element__label"
        htmlFor="small-id"
      >
        <span
          className="slds-slider-label"
        >
          <span
            className="slds-slider-label__label"
          >
            Slider Label
          </span>
          <span
            className="slds-slider-label__range"
          >
            0
             - 
            100
          </span>
        </span>
      </label>
      <div
        className="slds-form-element__control"
      >
        <div
          className="slds-slider slds-size_small"
        >
          <input
            className="slds-slider__range"
            id="small-id"
            max={100}
            min={0}
            onChange={[Function]}
            onInput={[Function]}
            step={1}
            type="range"
          />
          <span
            aria-hidden="true"
            className="slds-slider__value"
          >
            0
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="slds-col_padded"
  >
    <div
      className="slds-form-element"
    >
      <label
        className="slds-form-element__label"
        htmlFor="medium-id"
      >
        <span
          className="slds-slider-label"
        >
          <span
            className="slds-slider-label__label"
          >
            Slider Label
          </span>
          <span
            className="slds-slider-label__range"
          >
            0
             - 
            100
          </span>
        </span>
      </label>
      <div
        className="slds-form-element__control"
      >
        <div
          className="slds-slider slds-size_medium"
        >
          <input
            className="slds-slider__range"
            id="medium-id"
            max={100}
            min={0}
            onChange={[Function]}
            onInput={[Function]}
            step={1}
            type="range"
          />
          <span
            aria-hidden="true"
            className="slds-slider__value"
          >
            0
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="slds-col_padded"
  >
    <div
      className="slds-form-element"
    >
      <label
        className="slds-form-element__label"
        htmlFor="large-id"
      >
        <span
          className="slds-slider-label"
        >
          <span
            className="slds-slider-label__label"
          >
            Slider Label
          </span>
          <span
            className="slds-slider-label__range"
          >
            0
             - 
            100
          </span>
        </span>
      </label>
      <div
        className="slds-form-element__control"
      >
        <div
          className="slds-slider slds-size_large"
        >
          <input
            className="slds-slider__range"
            id="large-id"
            max={100}
            min={0}
            onChange={[Function]}
            onInput={[Function]}
            step={1}
            type="range"
          />
          <span
            aria-hidden="true"
            className="slds-slider__value"
          >
            0
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Sizes HTML Snapshot 1`] = `
"<div class=\\"slds-grid slds-grid_pull-padded slds-grid_vertical-align-center\\">
  <div class=\\"slds-col_padded\\">
    <div class=\\"slds-form-element\\"><label class=\\"slds-form-element__label\\" for=\\"x-small-id\\"><span class=\\"slds-slider-label\\"><span class=\\"slds-slider-label__label\\">Slider Label</span><span class=\\"slds-slider-label__range\\">0 - 100</span></span></label>
      <div class=\\"slds-form-element__control\\">
        <div class=\\"slds-slider slds-size_x-small\\"><input type=\\"range\\" id=\\"x-small-id\\" class=\\"slds-slider__range\\" min=\\"0\\" max=\\"100\\" step=\\"1\\" /><span class=\\"slds-slider__value\\" aria-hidden=\\"true\\">0</span></div>
      </div>
    </div>
  </div>
  <div class=\\"slds-col_padded\\">
    <div class=\\"slds-form-element\\"><label class=\\"slds-form-element__label\\" for=\\"small-id\\"><span class=\\"slds-slider-label\\"><span class=\\"slds-slider-label__label\\">Slider Label</span><span class=\\"slds-slider-label__range\\">0 - 100</span></span></label>
      <div class=\\"slds-form-element__control\\">
        <div class=\\"slds-slider slds-size_small\\"><input type=\\"range\\" id=\\"small-id\\" class=\\"slds-slider__range\\" min=\\"0\\" max=\\"100\\" step=\\"1\\" /><span class=\\"slds-slider__value\\" aria-hidden=\\"true\\">0</span></div>
      </div>
    </div>
  </div>
  <div class=\\"slds-col_padded\\">
    <div class=\\"slds-form-element\\"><label class=\\"slds-form-element__label\\" for=\\"medium-id\\"><span class=\\"slds-slider-label\\"><span class=\\"slds-slider-label__label\\">Slider Label</span><span class=\\"slds-slider-label__range\\">0 - 100</span></span></label>
      <div class=\\"slds-form-element__control\\">
        <div class=\\"slds-slider slds-size_medium\\"><input type=\\"range\\" id=\\"medium-id\\" class=\\"slds-slider__range\\" min=\\"0\\" max=\\"100\\" step=\\"1\\" /><span class=\\"slds-slider__value\\" aria-hidden=\\"true\\">0</span></div>
      </div>
    </div>
  </div>
  <div class=\\"slds-col_padded\\">
    <div class=\\"slds-form-element\\"><label class=\\"slds-form-element__label\\" for=\\"large-id\\"><span class=\\"slds-slider-label\\"><span class=\\"slds-slider-label__label\\">Slider Label</span><span class=\\"slds-slider-label__range\\">0 - 100</span></span></label>
      <div class=\\"slds-form-element__control\\">
        <div class=\\"slds-slider slds-size_large\\"><input type=\\"range\\" id=\\"large-id\\" class=\\"slds-slider__range\\" min=\\"0\\" max=\\"100\\" step=\\"1\\" /><span class=\\"slds-slider__value\\" aria-hidden=\\"true\\">0</span></div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`Vertical DOM Snapshot 1`] = `
<div
  className="slds-form-element"
>
  <label
    className="slds-form-element__label"
    htmlFor="vertical-id"
  >
    <span
      className="slds-slider-label"
    >
      <span
        className="slds-slider-label__label"
      >
        Slider Label
      </span>
      <span
        className="slds-slider-label__range"
      >
        0
         - 
        100
      </span>
    </span>
  </label>
  <div
    className="slds-form-element__control"
  >
    <div
      className="slds-slider slds-slider_vertical"
    >
      <input
        className="slds-slider__range"
        id="vertical-id"
        max={100}
        min={0}
        onChange={[Function]}
        onInput={[Function]}
        step={1}
        type="range"
      />
      <span
        aria-hidden="true"
        className="slds-slider__value"
      >
        0
      </span>
    </div>
  </div>
</div>
`;

exports[`Vertical HTML Snapshot 1`] = `
"<div class=\\"slds-form-element\\"><label class=\\"slds-form-element__label\\" for=\\"vertical-id\\"><span class=\\"slds-slider-label\\"><span class=\\"slds-slider-label__label\\">Slider Label</span><span class=\\"slds-slider-label__range\\">0 - 100</span></span></label>
  <div class=\\"slds-form-element__control\\">
    <div class=\\"slds-slider slds-slider_vertical\\"><input type=\\"range\\" id=\\"vertical-id\\" class=\\"slds-slider__range\\" min=\\"0\\" max=\\"100\\" step=\\"1\\" /><span class=\\"slds-slider__value\\" aria-hidden=\\"true\\">0</span></div>
  </div>
</div>"
`;
